hopsworks-api/src/main/java/io/hops/hopsworks/api/kafka/KafkaService.java

```diff
@@ -238,66 +244,95 @@ public Response topicIsSharedTo(@Context UriInfo uriInfo, @PathParam("topic") St
     return Response.ok().entity(dto).build();
   }
 
+  @ApiOperation(value = "Get all ACLs for a specified topic.")
   @GET
-  @Path("/aclUsers/topic/{topicName}")
+  @Path("/topics/{topic}/acl")
   @Produces(MediaType.APPLICATION_JSON)
   @AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
   @JWTRequired(acceptedTokens={Audience.API}, allowedUserRoles={"HOPS_ADMIN", "HOPS_USER"})
-  public Response aclUsers(@PathParam("topicName") String topicName) {
-    List<AclUserDTO> aclUsersDtos = kafkaFacade.aclUsers(project.getId(), topicName);
-    GenericEntity<List<AclUserDTO>> aclUsers = new GenericEntity<List<AclUserDTO>>(aclUsersDtos) {};
-    return Response.ok().entity(aclUsers).build();
+  public Response getTopicAcls(@Context UriInfo uriInfo,
+    @PathParam("topic") String topicName,
+    @BeanParam Pagination pagination,
+    @BeanParam AclsBeanParam aclsBeanParam) {
+    
+    ResourceRequest resourceRequest = new ResourceRequest(ResourceRequest.Name.KAFKA);
+    resourceRequest.setOffset(pagination.getOffset());
+    resourceRequest.setLimit(pagination.getLimit());
+    resourceRequest.setSort(aclsBeanParam.getSortBySet());
+    resourceRequest.setFilter(aclsBeanParam.getFilter());
+    AclDTO dto = aclBuilder.build(uriInfo, project, topicName, resourceRequest);
+    return Response.ok().entity(dto).build();
   }
-
+  
+  @GET
+  @Path("/topics/{topic}/users/acl")
```

Comment:
@kai-chi What does this endpoint do?

Reply:
we can redirect the traffic to `projectId/projectMembers` but the responses will be much bigger. just for 3 projects with two users the total difference is ~700B vs 10kB. is that of our concern? What could ease it a bit is an enpoint `projectId/projectMembers/emails` that returns only an array of emails. @tkakantousis, what do you think?